src/vs/platform/externalTerminal/node/externalTerminalService.ts

```diff
@@ -369,3 +375,152 @@ function quote(args: string[]): string {
 	}
 	return r;
 }
+
+namespace KittyTerminalService {
+	export interface KittyProcess {
+		cmdline: string[];
+		cwd: string;
+		pid: number;
+	}
+
+	export interface KittyWindow {
+		id: number;
+		title: string;
+		at_prompt: boolean;
+		foreground_processes: KittyProcess[];
+	}
+
+	export interface KittyTab {
+		windows: KittyWindow[];
+	}
+
+	export interface KittyInstance {
+		tabs: KittyTab[];
+	}
+}
+
+class KittyTerminalService {
```

Comment:
Since this a class that only has static members and never needs to be instantiated, let's move it to node/kitty.ts and export a single `runInKittyTerminal` function.